[LC-595] Edit a logic to validate an `unconfirmed_block` and both of `BlockVote` and `LeaderVote` include also validation the `round` and the generator ID.

hyerim-kim opened the pull request:
- Edit a logic to validate an `unconfirmed_block` and both of `BlockVote` and `LeaderVote` include also validation the `round` and the generator ID.
- Add `round` to the `BlockSend` and `LeaderVote` message.
- Add `__remove_outdated_complain_votes()` and apply after call `new_round()`.
- Edit a variable name to be short.
- Edit the related testcase.
- Remove some unnecessary logs and a message.

hyerim-kim committed:
[LC-595] Edit a logic to validate an `unconfirmed_block` includes also validation the `round`.

- Add `round` to the `BlockSend` message.
- Remove unnecessary message.

hyerim-kim committed:
[LC-595] Add the `round` to `LeaderVote` and `LeaderVotes`.

- Add `__remove_complain_votes()` and apply after call `new_round()`.
- Edit a variable name to be short.
- Remove some unnecessary logs.
- Edit the related testcase.

metalg0su commented on loopchain/blockchain/votes/v0_1a/vote.py:
```diff
@@ -69,36 +70,39 @@ def result(self) -> ExternalAddress:
     def _deserialize(cls, data: dict):
         data_deserialized = super()._deserialize(data)
         data_deserialized["block_height"] = int(data["blockHeight"], 16)
+        data_deserialized["round_"] = data["round_"]
```
Need to name a key with underbar_?

hyerim-kim replied:
The `round` is a reserved keyword in python builtin.

metalg0su replied:
I pointed that the name `round` is possible if it is used as a key of `dict`, not a bare usage.

```python
>>> a = {"round": 1}
>>> a["round"]
1
```

metalg0su commented on loopchain/blockchain/epoch.py:
```diff
@@ -103,14 +105,32 @@ def add_complain(self, leader_vote: LeaderVote):
         except RuntimeError as e:
             logging.warning(e)
 
+    def __remove_complain_votes(self):
```
I think this method aims to `__remove_outdated_complain_votes`

metalg0su commented on loopchain/peer/block_manager.py:
```diff
@@ -243,36 +243,52 @@ def confirm_prev_block(self, current_block: Block):
+            raise InvalidUnconfirmedBlock(
+                f"The unconfirmed block has invalid round. Expected({self.epoch.round}), Unconfirmed_block({round_})")
+
+        if self.epoch.round != round_:
```
Please check duplicated condition.

hyerim-kim committed:
[LC-595] Add the `round` to `BlockVote` and `BlockVotes`.

- Edit some conditions to validate the Epoch of an unconfirmed block.
- Rename a method.
- Edit some logs.
- Edit the related testcase.

hyerim-kim committed:
[LC-595] Add logic to vote for fail when a node gets the `ComplainLeader` message of a lower `round` than itself.

windies21 committed:
[LC-595] add find_preps_targets_by_roothash for get target by rep_id.
